foxtrot-core/src/main/java/com/flipkart/foxtrot/core/querystore/impl/ElasticsearchQueryStore.java

```diff
@@ -443,5 +436,74 @@ private String convert(Document translatedDocument) {
+     * @param table : TableV2 object containing all information related to templates for table
+     * @throws FoxtrotException
+     */
+    @Override
+    @Timed
+    public void initializeTable(TableV2 table) throws FoxtrotException {
+        logger.info("Starting template initialization for table {}", table.getTable().getName());
+
+        String tableName = table.getTable().getName();
+        PutIndexTemplateRequest putIndexTemplateRequest = getFoxtrotTableTemplateMappings(tableName, table.getTableTemplate());
+        PutIndexTemplateResponse response = connection.getClient()
+                .admin()
+                .indices()
+                .putTemplate(putIndexTemplateRequest)
+                .actionGet();
+        if (!response.isAcknowledged()) {
+            throw FoxtrotExceptions.createTableInitializationException(
+                    table.getTable(),
+                    "Template initialization failed. Check settings and mappings"
+            );
+        }
+
+        logger.info("Template initialization finished for table {}", tableName);
+    }
+
+    /**
+     * This function returns the PutIndexTemplateRequest for a given table.
+     * For ex : if table name is bro template created will be : template_foxtrot_bro_mappings
+     * This mapping will be applied to all indices with names of type foxtrot-bro--*
+     * @param table : Table name
+     * @param indexTemplate : index template containing settings and mappings that need to be put in template
+     * @return : PutIndexTemplateRequest object
+     * @throws FoxtrotException
+     */
+    private PutIndexTemplateRequest getFoxtrotTableTemplateMappings(String table, IndexTemplate indexTemplate) throws FoxtrotException {
+        Object settings = indexTemplate.getSettings();
+
+        Settings templateSettings = Settings.builder()
+                .put(settings)
+                .build();
+
+        String mappings;
+        try {
+            mappings = mapper.writeValueAsString(indexTemplate.getMappings());
+        } catch (JsonProcessingException e) {
+            logger.error("Json exception thrown in putIndexTemplateRequest by mapper : ", e.getMessage());
+            throw FoxtrotExceptions.createBadRequestException(table,
+                    String.format("Template initialization failed for table - %s due to json exception while parsing mappings.", table));
+        }
+        XContentBuilder templateMappings;
+        try {
+            XContentParser parser = XContentFactory.xContent(XContentType.JSON)
+                    .createParser(NamedXContentRegistry.EMPTY, mappings);
+
+            templateMappings = XContentFactory.jsonBuilder().copyCurrentStructure(parser);
+        } catch (Exception e) {
+            logger.error("Exception thrown in putIndexTemplateRequest : ", e.getMessage());
+            throw FoxtrotExceptions.createBadRequestException(table,
```

Comment:
Try relaying cause of parent exception. So that when you see this in the log, you realise what went wrong.